code/modules/mining/mining_encounters.dm

```diff
@@ -904,4 +916,13 @@
 			continue
 		AST.set_event(E)
 
+/proc/Turfspawn_Asteroid_SeedArtifacts(list/turfs)
+	var/turf/simulated/wall/auto/asteroid/AST
+
+	shuffle_list(turfs)
+
+	for (var/i in 1 to min(length(turfs), rand(7, 10)))
```

Comment:
Maybe move this rand out of the proc as an argument to align with the ore version?